scripts/js/lib/markdownTitles.ts

```diff
@@ -0,0 +1,64 @@
+import { unified } from "unified";
+import remarkParse from "remark-parse";
+import remarkGfm from "remark-gfm";
+import remarkFrontmatter from "remark-frontmatter";
+import { visit } from "unist-util-visit";
+import { Root } from "mdast";
+import yaml from "js-yaml";
+
+// Helper to recursively extract visible text from heading node
+function extractText(node: any): string {
+  if (node.type === "text" || node.type === "inlineCode") {
+    return node.value;
+  }
+
+  if (node.type === "link" && node.children) {
+    return node.children.map(extractText).join(" ");
+  }
+
+  if (node.children && Array.isArray(node.children)) {
+    return node.children.map(extractText).join(" ");
+  }
+
+  return "";
+}
+
+export async function collectHeadingTitleMismatch(
+  markdown: string,
+): Promise<string[]> {
```

Comment:
Thanks for switching to `Set`, however, I meant we should return a `Set` to match the interface of the images check.

```suggestion
): Promise<Set<string>> {
```

Then you can remove `Array.from` on line 63 and change the test results to be

```ts
const result: Set<string> = new Set(['...'])
```